future.go

```diff
@@ -15,32 +16,43 @@ type Future struct {
 	mutex     sync.Mutex
 	resp      Response
 	err       error
+	cond      sync.Cond
+	finished  atomic.Bool
```

Comment:
Right now `atomic.finished` is called under lock everywhere (except `isDone`), that is once or twice called without it. It's better to replace with simple `bool` calling it with lock, than making it atomic (that will invalidate cache record on every access to this variable).